### PR TITLE
Refactor Dockerfile

### PR DESCRIPTION
**Pull request by PeterDaveHello**

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@Neilpang would you like to take a look at this one? Thanks.

**Comment by Neilpang** [comment]
can you please give me more details about why it is necessary to do this refactor ?

**Comment by PeterDaveHello** [comment]
The reference is in the commit message, they are both in the official Dockerfile best practice documentation, and as you can see, same result, but shorter commands, easier to read.

**Comment by FernandoMiguel** [comment]
@neilpang you can add https://github.com/hadolint/hadolint to the ci and local to validate rules

**Comment by PeterDaveHello** [comment]
Thanks @FernandoMiguel :tada: 

@Neilpang I can help integrate linter with CI if you also prefer so ;)

**Comment by Neilpang** [comment]
@PeterDaveHello 
Sure, go ahead, please.

**Comment by PeterDaveHello** [comment]
I'll do it in another PR later when I have time, I assume this one could be merged first?

**Comment by Neilpang** [comment]
@PeterDaveHello 
I just took a closer look at the dockerfile lint project,   it seems overkill.
I don't think it necessary to add the lint tool to our CI process.
Our dockerfile is simple.

As for this PR, I need more time to review.

**Comment by PeterDaveHello** [comment]
Sure, official Dockerfile best practice will help, it simply indicates both of these usages ;)

**Comment by PeterDaveHello** [comment]
Ping @Neilpang for review, thanks.

**Comment by Neilpang** [comment]
please resolve the conflicts first.

Thanks.

**Comment by PeterDaveHello** [comment]
@Neilpang sure, it's done.

File: Dockerfile [hunk 1]
```diff
@@ -12,8 +11,7 @@ RUN apk update -f \
   tzdata \
   oath-toolkit-oathtool \
   tar \
-  libidn \
-  && rm -rf /var/cache/apk/*
```
Comment (Neilpang) [inline Dockerfile:16]: why is this line removed ?

Comment (FernandoMiguel) [reply Dockerfile:16]: newer docker images for most OSs do this via internal scripts.
